File: the-simple-webservice/python/the_simple_webservice/the_simple_webservice_stack.py
from aws_cdk import (
    aws_lambda as _lambda,
    aws_apigatewayv2 as api_gw,
    aws_apigatewayv2_integrations as integrations,
    aws_dynamodb as dynamo_db,
    core
)


class TheSimpleWebserviceStack(core.Stack):
    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # DynamoDB Table
        table = dynamo_db.Table(self, "Hits",
                                partition_key=dynamo_db.Attribute(name="path", type=dynamo_db.AttributeType.STRING)
                                )

        # defines an AWS  Lambda resource
        dynamo_lambda = _lambda.Function(self, "DynamoLambdaHandler",
                                            runtime=_lambda.Runtime.NODEJS_12_X,    # execution environment
                                            handler="lambda.handler",               # file is "lambda", function is "handler"
                                            code=_lambda.Code.from_asset("lambda_fns"), # Code loaded from the lambda_fns dir
                                            environment={
                                                'HITS_TABLE_NAME': table.table_name
                                            }
                                        )

        # grant the lambda role read/write permissions to our table'
        table.grant_read_write_data(dynamo_lambda)


        # defines an API Gateway Http API resource backed by our "dynamoLambda" function.
        api = api_gw.HttpApi(self, 'Endpoint', default_integration=integrations.LambdaProxyIntegration(handler=dynamo_lambda));
    
        core.CfnOutput(self, 'HTTP API Url', value=api.url);

File: the-simple-webservice/csharp/src/TheSimpleWebservice/TheSimpleWebserviceStack.cs
using System.Collections.Generic;
using Amazon.CDK;
using Amazon.CDK.AWS.APIGatewayv2;
using Amazon.CDK.AWS.APIGatewayv2.Integrations;
using Amazon.CDK.AWS.DynamoDB;
using Amazon.CDK.AWS.Lambda;

namespace TheSimpleWebservice
{
    public class TheSimpleWebserviceStack : Stack
    {
        internal TheSimpleWebserviceStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
        {
            //DynamoDB Table
            var table = new Table(this, "Hits", new TableProps
            {
                PartitionKey = new Attribute
                {
                    Name = "path",
                    Type = AttributeType.STRING
                }
            });
            
            // defines an AWS Lambda resource
            var dynamoLambda = new Function(this, "DynamoLambdaHandler", new FunctionProps
            {
                Runtime = Runtime.NODEJS_12_X,                // execution environment
                Code = Code.FromAsset("lambda-fns"),    //code loaded from the "lambda" directory
                Handler = "lambda.handler",                   //code loaded from the "lambda" directory
                Environment = new Dictionary<string, string>
                {
                    { "HITS_TABLE_NAME", table.TableName }
                }
            });
            
            // grant the lambda role read/write permissions to our table
            table.GrantReadWriteData(dynamoLambda);

            
            // defines an API Gateway Http API resource backed by our "dynamoLambda" function.
            var api = new HttpApi(this, "Endpoint", new HttpApiProps
            {
                DefaultIntegration = new LambdaProxyIntegration(new LambdaProxyIntegrationProps
                {
                    Handler = dynamoLambda
                })
            });

            new CfnOutput(this, "HTTP API Url", new CfnOutputProps
            {
                Value = api.Url ?? "Something went wrong with the deploy"
            });
        }
    }
}
